client/src/queries/tip.ts

```diff
@@ -0,0 +1,33 @@
+import { QueryFunction, queryOptions } from "@tanstack/react-query";
+import * as api from "./fetch/fetchWrapper";
+
+const fetchTip: QueryFunction<
+  UserArtistTip,
+  ["fetchTip", { artistId: string; artistSlug: string; tipId: string }]
+> = ({ queryKey: [_, { artistId, artistSlug, tipId }], signal }) => {
+  return api
+    .get<{
+      result: UserArtistTip;
+    }>(`/v1/artists/${artistId}/${artistSlug}/${tipId}`, {
+      signal,
+    })
+    .then((r) => r.result);
+};
+
+export function queryTip(opts: {
+  artistId: string;
+  artistSlug: string;
+  tipId: string;
+}) {
+  return queryOptions({
+    queryKey: [
+      "fetchTip",
+      {
+        artistId: opts.artistId,
+        artistSlug: opts.artistSlug,
+        tipId: opts.tipId,
+      },
+    ],
+    queryFn: fetchTip,
+  });
+}
```

Comment:
Using the idea that we'll implement an endpoint to fetch the id directly (as outlined above) `manage/tips/:tipId`

```suggestion
export function queryTip(opts: {
  tipId: string;
}) {
  return queryOptions({
    queryKey: [
      "fetchTip",
      {
        tipId: opts.tipId,
      },
    ],
    queryFn: fetchTip,
  });
}
```

Reply:
Resolved in [6ac18c4](https://github.com/funmusicplace/mirlo/pull/963/commits/6ac18c4c777ec8d847ee34ffe64e9879560cefb2)